src/gui/qgsmapcanvas.cpp

```diff
@@ -2796,6 +2796,15 @@ void QgsMapCanvas::dragEnterEvent( QDragEnterEvent *event )
   }
 }
 
+bool QgsMapCanvas::viewportEvent( QEvent *event )
+{
+  if ( event->type() == QEvent::ToolTip && mMapTool && mMapTool->canvasToolTipEvent( static_cast<QHelpEvent *>( event ) ) )
```

Comment:
```suggestion
  if ( event->type() == QEvent::ToolTip && mMapTool && mMapTool->canvasToolTipEvent( qobject_cast<QHelpEvent *>( event ) ) )
```
?

Reply:
Since the event type is being tested, static_cast should be safe. If one doesn't trust this, in addition to qobject_cast one should also test whether the resulting pointer is not null.

Reply:
Should actually be qgis::down_cast